cloud-controller-manager/do/resources.go

```diff
@@ -157,25 +154,17 @@ func (r *ResourcesController) syncTags() error {
 		return fmt.Errorf("failed to list load-balancers: %s", err)
 	}
 
-	loadBalancerIDsByName := make(map[string]string, len(lbs))
-	for _, lb := range lbs {
-		loadBalancerIDsByName[lb.Name] = lb.ID
-	}
-
 	// Collect tag resources for known load-balancers (i.e., services with
 	// type=LoadBalancer that either have our own LB ID annotation set or go by
 	// a matching name).
 	var res []godo.Resource
 	for _, svc := range lbSvcs {
-		id := getLoadBalancerID(svc)
-		if id == "" {
-			name := getDefaultLoadBalancerName(svc)
-			id = loadBalancerIDsByName[name]
+		id, err := r.resources.findLoadBalancerID(svc, lbs)
+		if err != nil && err != errLBNotFound {
+			return fmt.Errorf("error occurred when searching for a load balancer: %s", err)
 		}
 
-		// Renamed load-balancers that have no LB ID set yet would still be
-		// missed, so check again if we have found an ID.
-		if id != "" {
+		if len(id) > 0 {
```

Comment:
This change doesn't seem necessary. Can we undo it?

(FWIW, I prefer to use `len()` on data structures only and `!= ""` on strings because it makes it easier to see when a string is involved.)

Reply:
Sure, here it is just a matter of style, let's revert this change.